### PR TITLE
[CL-2] Move POL into nomos-core

### PR DESCRIPTION
**Pull request by zeegomo**
## 1. What does this PR implement?

Move PoL proof definition from ledger to nomos-core to uniform with the rest of the codebase. Also changes proof type from STARK to groth16. This currently requires an x86 machine with docker support (or a cloud provider), but it allows to reduce proof sizes consistenly (from 200kb to a few hundred bytes).

## 2. Does the code have enough context to be clearly understood?

Yes, mainly moved somewhere else.

## 3. Who are the specification authors and who is accountable for this PR?

No spec, myself.

## 4. Is the specification accurate and complete?
Not applicable
## 5. Does the implementation introduce changes in the specification?
Not applicable
## Checklist

* [x] 1. Description added.
* [x] 2. Context and links to Specification document(s) added.
* [x] 3. Main contact(s) (developers and specification authors) added
* [x] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 5. Link PR to a specific milestone.

### CODE REVIEW

File: nomos-core/chain-defs/src/header/leader_proof.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            .write(&public_inputs)
+            .unwrap()
+            .write(&private_inputs)
+            .unwrap()
+            .build()
+            .unwrap();
+
+        let start_t = std::time::Instant::now();
+
+        // ATTENTION: producing a groth16 proof currently requires x86 with docker support
```
Comment (danielSanchezQ) [inline nomos-core/chain-defs/src/header/leader_proof.rs:37]: They use virtualization for this? I'm really curious why is the case. 🤔

Comment (zeegomo) [reply nomos-core/chain-defs/src/header/leader_proof.rs:37]: Not sure why, I guess it's just to bundle dependencies for rapidsnark which is what they run inside the container

**Comment by danielSanchezQ** [comment]
Regarding the docker stuff. Should we wait till we have the CI ready for this? 
@bacv & @zeegomo ?

**Comment by bacv** [comment]
> Regarding the docker stuff. Should we wait till we have the CI ready for this? @bacv & @zeegomo ?

GH runners support running docker, we have docker-build job that checks if nomos image can be built. Or we could use the selfhosted linux runner with custom config tailored to this particular case.

**Review by danielSanchezQ — APPROVED** [review]

**Comment by zeegomo** [comment]
> Regarding the docker stuff. Should we wait till we have the CI ready for this?

AFAIK we were using dev mode before as well because it was too slow on GH runners. If we want to build proofs I think the only solution is self-hosted runners

**Review by davidrusu — APPROVED** [review]
🚢

**Pull request merged**